File: Lesson5 Sorting/Problem3/Solutions/Solution.java
import java.util.Scanner;
public class Solution {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt(); //number of test cases
		for (int i = 0; i < n; i++) { 
			int l = sc.nextInt, train = new int[l], swaps = 0; //l is the length of the array
			for (int j = 0; j < l; j++) {
				train[j] = sc.nextInt();
			}
			boolean isSorted = false; //boolean to check if the array is sorted
      			while(!isSorted){ //while the array is not sorted
       				isSorted = true;
				for (int j = 0; j < l - 1; j++) {
					if (train[j] > train[j + 1]) {
				    		int temp = train[j]; //bubble sort
						train[j] = train[j + 1];
						train[j + 1] = temp;
         					isSorted = false;
						swaps = swaps + 1; //every time we make a swap, we add 1 to the counter
					}
				}
			}
			System.out.println("Optimal train swapping takes " + swaps + " swap(s)."); //print the counter
		}
	}
}

File: Lesson5 Sorting/Problem3/Solutions/Solution.cpp
#include <iostream>

using namespace std;

int main()
{
    int n; cin>>n;
    while(n--){
        int l; cin>>l; int array[l];
        for(int i = 0; i < l; i++) {
            cin>>array[i];
        }
        int swaps = 0;
        bool isSorted = false;
        while(!isSorted){
            isSorted = true;
            for(int i = 0; i < l - 1; i++){
                if(array[i] > array[i + 1]){
                    int temp = array[i];
                    array[i] = array[i + 1];
                    array[i + 1] = temp;
                    swaps++;
                    isSorted = false;
                }
            }
        }
        cout<<"Optimal train swapping takes " <<swaps<<" swap(s)."<<endl;
    }
    return 0;
}
